src/modm/platform/rtc/stm32/rtc.hpp.in

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2023, Rasmus Kleist Hørlyck Sørensen
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_STM32_RTC_HPP
+#define MODM_STM32_RTC_HPP
+
+#include <chrono>
+
+#include <modm/platform/clock/rcc.hpp>
+#include <modm/math/algorithm/prescaler.hpp>
+#include <modm/architecture/interface/register.hpp>
+
+namespace modm::platform
+{
+
+/**
+ * Real Time Clock (RTC) control for STM32 devices
+ *
+ * @author Rasmus Kleist Hørlyck Sørensen
+ * @ingroup modm_platform_rtc
+ */
+class Rtc
+{
+    using duration = std::chrono::duration<float>;
```

Comment:
That is a hickup. I have been in doubt what type I should use since I wanted to use 64 bit, but it is only a 32 bit architecture. I have changed it to `int64_t`, which solves a lot of problems.